Firestore/Source/Remote/FSTSerializerBeta.mm

```diff
@@ -520,30 +522,30 @@ - (FSTMutation *)decodedMutation:(GCFSWrite *)mutation {
   }
 }
 
-- (GCFSPrecondition *)encodedPrecondition:(FSTPrecondition *)precondition {
-  FSTAssert(!precondition.isNone, @"Can't serialize an empty precondition");
+- (GCFSPrecondition *)encodedPrecondition:(const Precondition &)precondition {
+  FSTAssert(!precondition.IsNone(), @"Can't serialize an empty precondition");
   GCFSPrecondition *message = [GCFSPrecondition message];
-  if (precondition.updateTime) {
-    message.updateTime = [self encodedVersion:precondition.updateTime];
-  } else if (precondition.exists != FSTPreconditionExistsNotSet) {
-    message.exists = precondition.exists == FSTPreconditionExistsYes;
+  if (precondition.type() == Precondition::Type::UpdateTime) {
+    message.updateTime =
+        [self encodedVersion:static_cast<FSTSnapshotVersion *>(precondition.update_time())];
```

Comment:
Is this cast necessary? `-encodedVersion:` statically takes an `FSTSnapshotVersion` so the compiler should find the user-defined conversion automatically, no?

Reply:
done